tasks/ast_codegen/src/layout.rs

```diff
@@ -20,52 +24,63 @@ pub enum Layout {
+            static ref WELL_KNOWN_X32: HashMap<&'static str, Layout> =
+                HashMap::from([$((stringify!($typ), $layout_32)),*]);
+        }
     };
 }
 
-lazy_static! {
-    static ref WELL_KNOWN: HashMap<&'static str, Layout> = map! {
-        // builtins
-        bool => Layout::builtin::<bool>(),
-        u8 => Layout::builtin::<u8>(),
-        i8 => Layout::builtin::<i8>(),
-        u16 => Layout::builtin::<u16>(),
-        i16 => Layout::builtin::<i16>(),
-        u32 => Layout::builtin::<u32>(),
-        i32 => Layout::builtin::<i32>(),
-        u64 => Layout::builtin::<u64>(),
-        i64 => Layout::builtin::<i64>(),
-        u128 => Layout::builtin::<u128>(),
-        i128 => Layout::builtin::<i128>(),
-        f32 => Layout::builtin::<f32>(),
-        f64 => Layout::builtin::<f64>(),
-        char => Layout::builtin::<char>(),
-        &str => Layout::builtin::<&str>(),
-        // platform dependant
-        usize => Layout::Unknown,
-        isize => Layout::Unknown,
-        // well known types
-        // TODO: generate const assertions for these in the ast crate
-        Span => Layout::Layout { size: 8, align: 4 },
-        Atom => Layout::builtin::<&str>(),
-    };
+type Ptr32 = u32;
+
+well_known! {
+    // builtins
+    // types smaller or equal to 4 bytes are identical
+    bool => { 64: Layout::of::<bool>(), 32: Layout::of::<bool>()},
+    u8 => { 64: Layout::of::<u8>(), 32: Layout::of::<u8>() },
+    i8 => { 64: Layout::of::<i8>(), 32: Layout::of::<i8>() },
+    u16 => { 64: Layout::of::<u16>(), 32: Layout::of::<u16>() },
+    i16 => { 64: Layout::of::<i16>(), 32: Layout::of::<i16>() },
+    u32 => { 64: Layout::of::<u32>(), 32: Layout::of::<u32>() },
+    i32 => { 64: Layout::of::<i32>(), 32: Layout::of::<i32>() },
+    f32 => { 64: Layout::of::<f32>(), 32: Layout::of::<f32>() },
+    // alignment of these on 32bit machines are 4
+    u64 => { 64: Layout::of::<u64>(), 32: Layout::Layout { size: 8, align: 4 } },
+    i64 => { 64: Layout::of::<i64>(), 32: Layout::Layout { size: 8, align: 4 } },
+    f64 => { 64: Layout::of::<f64>(), 32: Layout::Layout { size: 8, align: 4 } },
+    u128 => { 64: Layout::of::<u128>(), 32: Layout::of::<u128>() },
+    i128 => { 64: Layout::of::<i128>(), 32: Layout::of::<i128>() },
+    char => { 64: Layout::of::<char>(), 32: Layout::of::<char>() },
+    &str => { 64: Layout::of::<&str>(), 32: Layout::of::<Ptr32>() },
+    // platform dependant
+    usize => { 64: Layout::of::<usize>(), 32: Layout::of::<u32>() },
+    isize => { 64: Layout::of::<isize>(), 32: Layout::of::<i32>() },
+    // well known types
+    // TODO: generate const assertions for these in the ast crate
+    Span => { 64: Layout::of::<oxc_span::Span>(), 32: Layout::of::<oxc_span::Span>() },
+    Atom => { 64: Layout::of::<oxc_span::Atom>(), 32: Layout::of::<oxc_span::Atom>() },
+    Vec => { 64: Layout::of::<oxc_allocator::Vec<'static, ()>>(), 32: Layout::of::<oxc_allocator::Vec<'static, ()>>()},
```

Comment:
To avoid having `oxc_allocator` as a dependency, I think we could hard-code this here as:

```rs
{
    64: Layout::Layout { size: 32, align: 8 },
    32: Layout::Layout { size: 16, align: 4 },
}
```

Eventually we should replace `Vec` with our own version (https://github.com/oxc-project/backlog/issues/18), at which point we can mark it `#[ast]` and parse it like other AST types.

Reply:
I've got rid of dependencies